distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/JvmOptionsParser.java

```diff
@@ -70,25 +71,27 @@ SortedMap<Integer, String> invalidLines() {
      * variable.
      *
      * @param args            the start-up arguments
-     * @param configDir       the ES config dir
+     * @param processInfo     info to pass through to the new Java subprocess
      * @param tmpDir          the directory that should be passed to {@code -Djava.io.tmpdir}
-     * @param envOptions      the options passed through the ES_JAVA_OPTS env var
      * @return the list of options to put on the Java command line
-     * @throws InterruptedException if the java subprocess is interrupted
-     * @throws IOException          if there is a problem reading any of the files
+     * @throws RuntimeException     if the java subprocess is interrupted
+     * @throws UncheckedIOException if there is a problem reading any of the files
      * @throws UserException        if there is a problem parsing the `jvm.options` file or `jvm.options.d` files
      */
-    static List<String> determineJvmOptions(ServerArgs args, Path configDir, Path tmpDir, String envOptions) throws InterruptedException,
-        IOException, UserException {
-
+    public static List<String> determineJvmOptions(ServerArgs args, ProcessInfo processInfo, Path tmpDir) throws UserException {
         final JvmOptionsParser parser = new JvmOptionsParser();
 
         final Map<String, String> substitutions = new HashMap<>();
         substitutions.put("ES_TMPDIR", tmpDir.toString());
-        substitutions.put("ES_PATH_CONF", configDir.toString());
+        substitutions.put("ES_PATH_CONF", args.configDir().toString());
+
+        final String envOptions = processInfo.envVars().get("ES_JAVA_OPTS");
 
         try {
-            return parser.jvmOptions(args, configDir, tmpDir, envOptions, substitutions);
+            List<String> jvmOptions = parser.jvmOptions(args, args.configDir(), tmpDir, envOptions, substitutions);
+            // also pass through distribution type
+            jvmOptions.add("-Des.distribution.type=" + processInfo.sysprops().get("es.distribution.type"));
```

Comment:
Perhaps this could be added to SystemJvmOptions (adding ProcessInfo as an arg to that method)?

Reply:
Done!